### PR TITLE
Add initial version of HTS locking HIP

### PR DESCRIPTION
**Pull request by sam-at-toko**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by sam-at-luther** [comment]
fyi @atul-hedera

**Comment by web3-nomad** [comment]
another way of achieving this is implemented in the cosmos SDK where they have created a 'vesting' account.. so the logic would be at the account level instead of the token one.

https://docs.cosmos.network/v0.45/modules/auth/05_vesting.html

**Comment by sam-at-luther** [comment]
I've added a discussion [here](https://github.com/hashgraph/hedera-improvement-proposal/discussions/759).

**Comment by sam-at-luther** [comment]
> another way of achieving this is implemented in the cosmos SDK where they have created a 'vesting' account.. so the logic would be at the account level instead of the token one.
> 
> https://docs.cosmos.network/v0.45/modules/auth/05_vesting.html

This iss along the lines of the “VestingWallet” discussed in the "Feature Support in other Languages" section.

**Review by eleonora-Toko — APPROVED** [review]
Initial version of HIP proposing a lockable functionality on amounts of fungible tokens 🎉.

File: HIP/hip-751.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+I WANT to have the option to enable the transfer lock function for a portion of my fungible token supply with a predetermined execution time,
+
+SO THAT I can schedule the unlocking of tokens in advance without the need for immediate manual execution.
+
+**_Scenario_** -  _Token Issuer_: Executing native events on the locked portion of a token supply
+
+AS A _Token Issuer_,
+
+I WANT to be able to execute events (metadata update, mint, burn, transfers and more) on my token even when the lock functionality is enabled,
```
Comment (mhess-swl) [inline HIP/hip-751.md:143]: For clarification, it sounds like a token _issuer_ can still transfer locked tokens, but token _holders_ can't exchange these locked tokens. Did I get that right?

Comment (eleonora-Toko) [reply HIP/hip-751.md:143]: @mhess-swl - Yes, that is correct. The owner of the LockKey can execute events on the tokens even while they are in a locked state, including transfers.  The lock primitive, as we envision it, targets a specific account, the account that should be the token holder will not be able to successfully execute the transfer till the lock has expired.

Comment (mhess-swl) [reply HIP/hip-751.md:143]: Understood, thank you!

**Comment by Stephanie-Yi** [comment]
Replaced by https://github.com/hashgraph/hedera-improvement-proposal/pull/796.

**Comment by mgarbs** [comment]
This hip was rewritten into 796 and since it was never merged we dont need it

**Pull request closed without merging**